pkg/reconciler/pipelinerun/pipelinerun.go

```diff
@@ -315,7 +329,19 @@ func (c *Reconciler) resolvePipelineState(
 		// We need the TaskRun name to ensure that we don't perform an additional remote resolution request for a PipelineTask
 		// in the TaskRun reconciler.
 		trName := resources.GetTaskRunName(pr.Status.TaskRuns, pr.Status.ChildReferences, task.Name, pr.Name)
-		fn, err := tresources.GetTaskFunc(ctx, c.KubeClientSet, c.PipelineClientSet, c.resolutionRequester, pr, task.TaskRef, trName, pr.Namespace, pr.Spec.ServiceAccountName)
+
+		// get verificationpolicies for trusted resources
+		cfg := config.FromContextOrDefaults(ctx)
+		var verificationpolicies []*v1alpha1.VerificationPolicy
+		if cfg.FeatureFlags.ResourceVerificationMode == config.EnforceResourceVerificationMode || cfg.FeatureFlags.ResourceVerificationMode == config.WarnResourceVerificationMode {
+			var err error
+			verificationpolicies, err = c.verificationPolicyLister.VerificationPolicies(pr.Namespace).List(labels.Everything())
+			if err != nil {
+				return nil, err
+			}
+		}
+
```

Comment:
I think you are running the same logic likes 333 -> 343 and 190 -> 200 multiple times. Might make sense to wrap it into a function?

Reply:
sure! Thanks!